src/gasman.c

```diff
@@ -1955,7 +1955,16 @@ UInt CollectBags (
         // extract the head from the linked list
         first = MarkedBags;
         MarkedBags = LINK_BAG(first);
-        LINK_BAG(first) = MARKED_ALIVE(first);
+        // gasman is inconsistent if a bag starting at YoungBags is
+        // a young bag. However at present such bags cannot exist.
+        // Add a check to ensure this property never changes accidentally.
```

Comment:
When we construct a bag, we always allocate a header just before the bag data, so the first bag after a GC is a little further along. This could break if anyone started rearranging bag internal structure.

This is a weird bit of code. I realised we were inconsistent with how we handled a bag at YoungBags, worried about it, then realised it can't actually happen. This check is mainly here so if anyone makes bags which can be at YoungBags, they'll trigger this and think about it :)